src/services/device-management/iot-device.service.ts

```diff
@@ -467,15 +468,54 @@ export class IoTDeviceService {
     }
 
     async delete(device: IoTDevice): Promise<DeleteResult> {
-        if (device.type == IoTDeviceType.LoRaWAN) {
-            const lorawanDevice = device as LoRaWANDevice;
-            this.logger.debug(
-                `Deleting LoRaWANDevice ${lorawanDevice.id} / ${lorawanDevice.deviceEUI} in Chirpstack ...`
-            );
-            await this.chirpstackDeviceService.deleteDevice(lorawanDevice.deviceEUI);
+        let deleteResult: DeleteResult = {
+            raw: null,
+            affected: 0
         }
 
-        return this.iotDeviceRepository.delete(device.id);
+        // If operations on chirpstack fail, rollback any mutations on the database
+        await getManager().transaction((async transactionManager => {
```

Comment:
Are we also handling the issue with the datatarget? As far as I can see we are only handling the case with multicast.

Reply:
Right before deleting the device, we set the data targets, `connections`, to `[]` and save it.